### PR TITLE
[FSR] Updating redux with cash comparison boolean

### PR DESCRIPTION
**Pull request by Dr-Pongo**
## Summary

**Problem:** Cash on hand streamlined page is showing the limited chapter list when it should have the full length. `isStreamlinedShortForm` is checking the cash on hand value and triggering an true before the value is confirmed. 

**Solution:** Set a cash on hand specific value for `isStreamlinedShortForm` to check when updating the cash on hand value.

**Note:** _Also sneaking in a fix so the GMT data api call won't tigger if the streamlined flag is disabled_

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#64263

## Testing done
Local testing complete, automated tests passing

## Screenshots

_Note: This is when the veteran first enters the page, if they navigate past and go back it will reflect the current state and change when the value is updated and navigation is selected._

| Before | After |
| ------ | ----- |
| image | image |

## What areas of the site does it impact?
FSR Streamlined waiver
*(Describe what parts of the site are impacted **if** code touched other areas)*

## Acceptance criteria
- [x] Cash on hand shows correct chapter numbers for state
### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Review by OptionSelect — APPROVED** [review]
LGTM! 🚀

File: src/applications/financial-status-report/components/monetary/CashOnHand.jsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import FormNavButtons from 'platform/forms-system/src/js/components/FormNavButto
 import { VaNumberInput } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
 import { isValidCurrency } from '../../utils/validations';
 import { currency as currencyFormatter } from '../../utils/helpers';
+import { safeNumber } from '../../utils/calculateIncome';
```
Comment (amponce) [inline src/applications/financial-status-report/components/monetary/CashOnHand.jsx:8]: making a mental note to change this to utils/helpers before caclculateExpenses gets merged

File: src/applications/financial-status-report/components/monetary/CashOnHand.jsx [hunk 2]
```diff
@@ -33,6 +34,16 @@ const CashOnHand = ({
...
+
+    // update gmtData for isStreamlinedShortForm
+    setFormData({
+      ...data,
+      gmtData: {
+        ...gmtData,
+        cashBelowGmt: safeNumber(cash) < gmtData?.assetThreshold,
+      },
+    });
+
```
Comment (amponce) [inline src/applications/financial-status-report/components/monetary/CashOnHand.jsx:46]: maybe we can combine this into a single setFormData?

```
setFormData({
  ...data,
  assets: {
    ...data.assets,
    cashOnHand: cash,
  },
  gmtData: {
    ...gmtData,
    cashBelowGmt: safeNumber(cash) < gmtData?.assetThreshold,
  },
});
```

**Review by amponce — APPROVED** [review]

File: src/applications/financial-status-report/components/monetary/CashOnHand.jsx (hunk 2 above)
Comment (Dr-Pongo) [reply src/applications/financial-status-report/components/monetary/CashOnHand.jsx]: Updated accordingly! Thanks for the input!

**Pull request merged**